sqlalchemy_redshift/__init__.py

```diff
@@ -21,7 +21,15 @@
 
 from sqlalchemy.dialects import registry
 
-registry.register("redshift", "sqlalchemy_redshift.dialect", "RedshiftDialect")
 registry.register(
-    "redshift.psycopg2", "sqlalchemy_redshift.dialect", "RedshiftDialect"
```

Comment:
For users directly using `RedshiftDialect` class, this would be a breaking change. Searching GitHub, I found a number of projects reference this class directly who would impacted. [see here](https://github.com/search?l=Python&q=RedshiftDialect&type=Code). So we probably want to take a different approach here.

One option to avoid a breaking change here could be to re-add `RedshiftDialect` having it inherit from `Pyscopg2RedshiftDialect`. Or we simply re-name `Pyscopg2RedshiftDialect` -> `RedshiftDialect`.

Reply:
I do like modernizing the names here, so I think it's good to add some shim for compatibility rather than staying stuck with the old name.

I think we can get away with something like this in `dialect.py`:

```
# Add RedshiftDialect synonym for backwards compatibility.
RedshiftDialect = RedshiftDialect_psycopg2
```

Reply:
Addressed in 291d88e7